kore/test/Test/Kore/Step/Rule/Combine.hs

```diff
@@ -379,11 +380,20 @@ applyAlias name aliasRight =
             }
         []
 
+runMergeRulesNoSMT
+    :: [RewriteRule VariableName]
+    -> IO [RewriteRule VariableName]
+runMergeRulesNoSMT (rule : rules) =
+    runNoSMT
+    $ runSimplifier Mock.env
+    $ mergeRules (rule :| rules)
+runMergeRulesNoSMT [] = error "Unexpected empty list of rules."
+
 runMergeRules
     :: [RewriteRule VariableName]
     -> IO [RewriteRule VariableName]
 runMergeRules (rule : rules) =
-    runNoSMT
+    runSMT (pure ())
```

Comment:
Why doesn't this need the declarations from `Mock.env`?

Reply:
I didn't investigate in depth, but it might be that it uses `runSimplifier Mock.env` on the next line

Reply:
I didn't investigate in depth, but i think it might be that it uses `runSimplifier Mock.env` below